stream-chat-android-client/src/main/java/io/getstream/chat/android/client/network/NetworkStateProvider.kt

```diff
@@ -55,6 +55,10 @@ internal class NetworkStateProvider(private val connectivityManager: Connectivit
     }
 
     fun subscribe(listener: NetworkStateListener) {
+        if (listeners.contains(listener)) {
```

Comment:
You can change the type of this container to `Set` and avoid manual check

Reply:
This is just a raw PR to showcase the issue. Also `listeners + listener` internally creates a new collection.